### ISSUE TITLE
Lettuce instrumentation does not sanitize password in exception message 

### ISSUE THREAD

**Issue by ericmm** [open-telemetry/opentelemetry-java-instrumentation#8422]
**Describe the bug**
The value of `db.statement` gets sanitized properly thanks to `RedisCommandSanitizer`. However when there's a RedisCommandExecutionException thrown, the exception message, stacktrace & status description may contain sensitive data (e.g. password), and they get logged in plain text. 

code here: 
https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/instrumentation/lettuce/lettuce-5.1/library/src/main/java/io/opentelemetry/instrumentation/lettuce/v5_1/OpenTelemetryTracing.java#LL235C31-L235C31

and here:
https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/instrumentation/lettuce/lettuce-5.1/library/src/main/java/io/opentelemetry/instrumentation/lettuce/v5_1/OpenTelemetryTracing.java#L242

pic1:
![otel-redis-error1](https://user-images.githubusercontent.com/1488675/236403025-75292693-05f6-45b7-ad4f-d2b20b396207.png)

pic2:
![otel-redis-error2](https://user-images.githubusercontent.com/1488675/236403065-faaba6ad-e5a6-413c-a12a-edf2302461e6.png)

pic3: 
![otel-redis-error3](https://user-images.githubusercontent.com/1488675/236403086-627dfeef-2cbc-4ada-a2aa-79d1c41a68e9.png)

**Steps to reproduce**
N.A.

**What did you expect to see?**
the sensitive data should be redacted 

**What did you see instead?**
the sensitive data is in plain text.

**What version are you using?**
latest version

**Environment**
Compiler:  Jdk17
OS: Linux

**Additional context**
Checked the Lettuce core source code, it does not have special handling on the error output. The error message ("ERR unknown command `HELLO`, with args beginning with: `3`, `AUTH`, `xxxx`, `xxxxxxx`, ") seems directly from Redis server response.

**Comment by mateuszrzeszutek** [comment]
Also relevant: #3039

**Comment by trask** [comment]
hi @ericmm, it may also be worth asking in the lettuce repo in case there's anything they can do without limiting the usefulness of the exception messages, since this would affect other folks who are logging these exceptions

**Comment by ericmm** [comment]
Thanks @trask & @mateuszrzeszutek , the error message seems directly from Redis server (https://github.com/redis/redis/blob/07d54dc5baa1b2e7adb5ce8e7bee398d376fe13b/src/server.c#L3729). I think the problem here is users have no control over this logging since it's done at instrumentation level. If it's a lettuce (or other redis library) exception thrown in users' code, they can choose to wrap it with another exception (not exposing the original stacktrace).

**Comment by trask** [comment]
@maryliag did you end up handling a similar issue in Node.js?

**Comment by maryliag** [comment]
Yes, on Node.js this is no sanitization on the error message, so sensitive information could leak. Since there was not easy/performant way to do this, I was making the change to remove to not save the error message itself, and use just the code for the error type
